nlprule/src/bin/compile.rs

```diff
@@ -1,9 +1,14 @@
 use clap::Clap;
 use nlprule::compile::{compile, BuildOptions, Error};
+use std::io::BufWriter;
+use fs_err as fs;
 
 fn main() -> Result<(), Error> {
     env_logger::init();
     let opts = BuildOptions::parse();
 
-    compile(&opts)
+    let tokenizer_sink = BufWriter::new(fs::File::create(&opts.tokenizer_out)?);
+    let rules_sink = BufWriter::new(fs::File::create(&opts.rules_out)?);
+
+    compile(opts.build_dir, rules_sink, tokenizer_sink)
```

Comment:
This change is good, can you also move the definition of `BuildOptions` to `bin/compile.rs`? Pretty sure it is only used here now.